test/proxyprotocol/proxyprotocol_test.go

```diff
@@ -91,6 +91,77 @@ func TestProxyProtocol(t *testing.T) {
 	assert.Equal(t, resp.StatusCode, http.StatusOK)
 }
 
+// TestProxyProtocolWithSNI verifies that the kourier is configured with proxy protocol and SNI.
+func TestProxyProtocolWithSNI(t *testing.T) {
+	ctx, clients := context.Background(), test.Setup(t)
+	privateHostname := test.ObjectNameForTest(t) + "." + test.ServingNamespace + ".svc." + test.NetworkingFlags.ClusterSuffix
+	name, port, _ := ingress.CreateProxyService(ctx, t, clients, privateHostname, "kourier-internal.kourier-system.svc.cluster.local")
```

Comment:
```suggestion
        name, port, _ := ingress.CreateProxyService(ctx, t, clients, privateHostname, "kourier-internal.kourier-system.svc."+test.NetworkingFlags.ClusterSuffix)
```

kind test does not use `cluster.local` so we should not use `clsuter.local`. https://github.com/knative-sandbox/net-kourier/blob/main/.github/workflows/kind-e2e.yaml#L55-L57